.github/workflows/_shared-docs-build-pr.yml

```diff
@@ -287,7 +287,12 @@ jobs:
       - name: Checkout BASE
         uses: actions/checkout@v3
         with:
-          ref: ${{ github.event.pull_request.base.sha }}
+          # Use base.ref, not base.sha, to make sure we do not get a checkout of the base at an earlier time
+          # (when the PR's branch was branched off the PR's target branch), but right now. Without this there
+          # will be a potentially large list of changes since the merge commit we check out for HEAD is based
+          # on a far newer version. I'm not 100% sure whether this is also a good idea for the closed event
+          # though...
```

Comment:
https://github.com/ansible-collections/community.docker/actions/runs/4085894401/jobs/7044478553 is a build after merging a PR. The bot comment was removed from https://github.com/ansible-collections/community.docker/pull/574.

Reply:
> One thing I'm not sure about is whether the diff is actually needed after closing or merging the PR.

It's needed in order to provide a unique "closed"/"merged" event comment that only gets applied when there was a difference (as opposed to always posting/editing it). Without the diff, we'd either be adding a comment always, or we could opt out of changing the comment on close/merge.

Perhaps an alternative to calculating diffs on close/merge is to skip all builds, and only update a comment when one exists already.

IIRC, the logic and hoops we have to jump through in the workflows end up being more complicated but that might be worth it to 1) save CI time and 2) if it removes some events that don't have the ref we want.

Reply:
I think avoiding the 'final' run of the docs build would be good, it saves a lot of CI resources (especially on large collections) and right now is the main problem for this PR, as I don't see a good way to do this correctly.

Reply:
I think I agree.. I don't want to lose the functionality, so checking for an existing comment instead of building is a good compromise I think. The only question is whether that is the only thing preventing us from using a better ref, or if there's something else I forgot...

Reply:
My comment here suggests that indeed only close is what prevents us from using the merge commit:
https://github.com/ansible-community/github-docs-build/pull/38
> Changing to use the merge branch in all cases except on PR close where it's not available.

Reply:
actually now I am wondering whether that's even the same thing as this... I think I am confusing myself 😖 

[to quote you](https://github.com/ansible-community/github-docs-build/pull/14#issuecomment-1008309523):
> Bah, this is really a huge mess.